### PR TITLE
deprecate_unused_legacy_vote_plumbing and timely_vote_credits

### PR DESCRIPTION
**Pull request by jumpsiegel**

### CODE REVIEW

File: src/flamenco/runtime/program/fd_vote_program.c [hunk 1]
```diff
@@ -1303,42 +1359,53 @@ process_new_vote_state( fd_vote_state_t *           vote_state,
...
     vote_state->last_timestamp.timestamp = *timestamp;
   }
+
+  // https://github.com/anza-xyz/agave/blob/v2.0.0/programs/vote/src/vote_state/mod.rs#L754
   vote_state->has_root_slot = (uchar)has_new_root;
   vote_state->root_slot     = new_root;
-  // TODO can prob just fd_memcpy
+
+#if 0
+  // WTF
```
Comment (jumpsiegel) [inline src/flamenco/runtime/program/fd_vote_program.c:1399]: I cannot find this code anywhere... went all the way back to 1.14.x ... why was this here?!

Comment (topointon-jump) [reply src/flamenco/runtime/program/fd_vote_program.c:1399]: https://github.com/anza-xyz/agave/blob/v2.0.0/programs/vote/src/vote_state/mod.rs#L755

**Review by topointon-jump — APPROVED** [review]
